R/sce_to_anndata.R

```diff
@@ -46,12 +46,23 @@ sce_to_anndata <- function(sce, anndata_file, x_assay_name = "counts") {
   # assign SCE to new variable to avoid modifying input SCE
   sce_to_convert <- sce
 
-  # remove miQC model from metadata
-  if (!is.null(metadata(sce_to_convert)$miQC_model)) {
-    metadata(sce_to_convert)$miQC_model <- NULL
-    message("miQC model cannot be converted between SCE and AnnData.")
+  # grab existing metadata
+  metadata_list <- metadata(sce_to_convert)
+
+  # remove any objects or dataframes
+  metadata_to_keep <- metadata_list |>
+    purrr::discard(is.object) |>
+    purrr::discard(is.data.frame)
```

Comment:
I worry about lists too, and data frames are lists, so this captures both.
```suggestion
    purrr::discard(is.list)
```